### DISCUSSION TITLE
Getting more info for `z` menu

### DISCUSSION THREAD

**Discussion by Archie-2021** [gokcehan/lf#1415] [Q&A] [answered]
I want to add more info about files. The basic Time and Size is not enough. I am in lf most of the times instead of terminal now a days and I want to see something like ls output of commands in lf (of course I want it to be interactive). Is there a config setup you recommend for this? Where should I look for such a thing.

**Comment by joelim-work** [comment]
You are looking for the `info` option, although currently there is only support for `size`, `time`, `atime`, and `ctime`. If you want other fields to be displayed (e.g. file permissions and ownership), you can create an issue to request this, although I think most users just look at the stat info in the bottom left corner.

**Reply by Archie-2021** [reply]
I now recall I had made a similar topic before here. What I am looking for is to have some sort of `ls` view inside `lf` I know I can see it in status bar but seeing all the files at once is another story. I mean I already using lf to see the files instead of ls and I am using `lfcd` to change directory. It makes sense to have the ls inside lf too as they don't take that much more space.

**Reply by gokcehan** [reply]
@Archie-2021 Most people have multiple columns in their ui and there is often not much space left to display too much information, so it is a little complicated for `lf`. We also still lack a mechanism to control the display of `info` as we only have an automatic display mechanism when the information size is less than half the width of the column. As a workaround, maybe you can define a custom command like `ls -l | less` with a mapping so you can switch to this display whenever you need to see all available information.

**Reply by Archie-2021** [reply]
> @Archie-2021 Most people have multiple columns in their ui and there is often not much space left to display too much information, so it is a little complicated for `lf`. We also still lack a mechanism to control the display of `info` as we only have an automatic display mechanism when the information size is less than half the width of the column. As a workaround, maybe you can define a custom command like `ls -l | less` with a mapping so you can switch to this display whenever you need to see all available information.

yeah. i thought about that. it is not a bad solution but still it would be cool to have all the info live.

**Reply by gokcehan** [reply]
@Archie-2021 I'm not against adding more information to `info` either. Feel free to open issue(s) to track it, if there isn't any issues yet. One thing to be careful is that I think it should somehow be possible to have an equal width for the information so they would align properly. For example, `size` is currently always a width of 4, and `time` is currently a width of 12 with the default settings. It is easy to have permissions with a width of 10 but other information (i.e. link count, user, group) can be a little tricky and we may need to figure out a way to have a predetermined width for these fields.

**Reply by Archie-2021** [reply]
> @Archie-2021 I'm not against adding more information to `info` either. Feel free to open issue(s) to track it, if there isn't any issues yet. One thing to be careful is that I think it should somehow be possible to have an equal width for the information so they would align properly. For example, `size` is currently always a width of 4, and `time` is currently a width of 12 with the default settings. It is easy to have permissions with a width of 10 but other information (i.e. link count, user, group) can be a little tricky and we may need to figure out a way to have a predetermined width for these fields.

Yes some of info like you could be problematic. I think we could start with permissions first and add other stuff later. I think permissions would be most useful one anyway.
I will try to open up a FR for that.

Thanks for your helo

**Comment by CatsDeservePets** [comment, marked as answer]
As of today, `info` now supports all of these:
```
perm      file permission
user      user name
group     group name
size      file size
time      time of last data modification
atime     time of last access
btime     time of file birth
ctime     time of last status (inode) change
custom    property defined via `addcustominfo` (empty by default)
```

The most interesting might be `custom` as it can be basically anything you want. You see multiple example usage cases mentioned in the [documentation] (https://github.com/gokcehan/lf/blob/master/doc.md#loading-directory) (e.g. `Git information`, `image resolution`). It can also be used for user-defined sorting.
